pkgs/applications/networking/browsers/w3m/default.nix

```diff
@@ -1,16 +1,18 @@
 { stdenv, fetchurl
-, sslSupport ? true
-, graphicsSupport ? false
-, mouseSupport ? false
-, ncurses, openssl ? null, boehmgc, gettext, zlib
-, imlib2 ? null, xlibsWrapper ? null, fbcon ? null
-, gpm-ncurses ? null
+, ncurses, boehmgc, gettext, zlib
+, sslSupport ? true, openssl ? null
+, graphicsSupport ? true, imlib2 ? null
+, x11Support ? graphicsSupport, libX11 ? null
```

Comment:
`LIB` thingy is okay, yep, I'll fix that.

`graphicsSupport` thingy in your comment and PR is not okay, because my
patch also works without X11 in frame buffer (I actually use that with
`ranger`).

I also don't like the idea of `w3m == w3m-pure` because then `ranger`,
`emacs` and everything else that uses `w3m` as a browser (that is, all
user-oriented stuff) will need to be overriden which is not nice. I can
agree to rename `w3m-pure` to `w3mBatch` and define `w3mFull = w3m` if
that bothers anyone, but nothing less.

I say after I fix `LIB` thingy, test that it doesn't break the frame
buffer and everyone agrees to these changes, then you should apply your
`fetchpatch`-changes (on top or in parallel, shouldn't matter, to this).